scripts/format_sources.sh

```diff
@@ -0,0 +1,73 @@
+#!/bin/bash
+
+#
+# Copyright (C) 2016-2017 Pelagicore AB
+#
+# Permission to use, copy, modify, and/or distribute this software for
+# any purpose with or without fee is hereby granted, provided that the
+# above copyright notice and this permission notice appear in all copies.
+#
+# THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL
+# WARRANTIES WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED
+# WARRANTIES OF MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR
+# BE LIABLE FOR ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES
+# OR ANY DAMAGES WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS,
+# WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION,
+# ARISING OUT OF OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS
+# SOFTWARE.
+#
+# For further information see LICENSE
+#
+
+#
+# This is a helper script for invoking uncrustify on a set of C++ files.
+# Invoking this script on a directory containing C++ files (*.cpp, *.h, *.c)
+# will format these files _in place_, according to an uncrustify config file.
+#
+# The $UNCRUSTIFY_CONFIG_FILE environment variable can be used to specify which
+# uncrustify config file to use.
+#
+
+COMPONENT_FOLDER="" # Set to softwarecontainer base dir
+UNCRUSTIFY_CONFIG_FILE="" # Add option to set this
+
+UNCRUSTIFY_LOGFILE="uncrustify_output.log" # Currently not used
+
+declare -a dirs=("agent" "common" "libsoftwarecontainer")
+
+if [ "$COMPONENT_FOLDER" = "" ];
+then
+  COMPONENT_FOLDER=`pwd`
+fi
+
+if [ "$UNCRUSTIFY_CONFIG_FILE" = "" ];
+then
+  UNCRUSTIFY_CONFIG_FILE=`dirname $0`/uncrustify.cfg
+fi
+
+echo "Formatting with Uncrustify configuration file : $UNCRUSTIFY_CONFIG_FILE"
+echo "Running uncrustify starting at $COMPONENT_FOLDER"
+
+## This will create a dir formatting_ouput with all the uncrustify result files
+UNCRUSTIFY_COMMAND="uncrustify -c $UNCRUSTIFY_CONFIG_FILE --prefix formatting_output 'file'"
+
+read -p "We are going to format the C/C++ (*.cpp, *.c, *.h) source files from the folder
+'$COMPONENT_FOLDER'. We recommend that you first commit your files to your local GIT repo
+before going further, since the formatter will modify your files. You will be able to amend
```

Comment:
Same as above. I forgot to fix the comment.
Also: when we change to use this with make or in the CI system we probably don't want this prompt, since it waits for an answer.